File: backend/src/routes/followRoutes.js
const express = require('express');
const router = express.Router();
const {
  followUser,
  unfollowUser,
  getFollowers,
  getFollowing,
  checkFollowStatus,
  getFollowStats
} = require('../controllers/followController');
const { authenticateToken } = require('../middleware/auth');

// All routes require authentication
router.use(authenticateToken);

// Follow/unfollow a user
router.post('/:userId/follow', followUser);
router.delete('/:userId/unfollow', unfollowUser);

// Get followers and following
router.get('/:userId/followers', getFollowers);
router.get('/:userId/following', getFollowing);

// Check follow status
router.get('/:userId/status', checkFollowStatus);

// Get follow stats
router.get('/:userId/stats', getFollowStats);

module.exports = router;

File: src/GlobalScout/GlobalScout.Api/Endpoints/Social/Follow/FollowRoutes.cs
namespace GlobalScout.Api.Endpoints.Social.Follow;

internal static class FollowRoutes
{
    public const string Base = "api/follow";

    public static string FollowUser => $"{Base}/{{userId:guid}}/follow";

    public static string UnfollowUser => $"{Base}/{{userId:guid}}/unfollow";

    public static string Followers => $"{Base}/{{userId:guid}}/followers";

    public static string Following => $"{Base}/{{userId:guid}}/following";

    public static string Status => $"{Base}/{{userId:guid}}/status";

    public static string Stats => $"{Base}/{{userId:guid}}/stats";
}

internal static class FollowEndpointTags
{
    public const string Follow = "Follow";
}
